wolfcrypt/src/asn.c

```diff
@@ -8909,9 +8909,9 @@ int wc_DerToPemEx(const byte* der, word32 derSz, byte* output, word32 outSz,
 
     /* build header and footer based on type */
     XSTRNCPY(header, headerStr, headerLen - 1);
-    header[headerLen - 1] = 0;
+    header[headerLen - 2] = 0;
     XSTRNCPY(footer, footerStr, footerLen - 1);
-    footer[footerLen - 1] = 0;
+    footer[footerLen - 2] = 0;
```

Comment:
This change (and above) will truncate the last byte if the header string size is `headerLen - 1`

Reply:
Yes, if the buffer size is too small. The point of this change is to prevent writing beyond the buffer with the `XSTRNCAT`s below.

Reply:
OK, thank you for clarifying.